public_html/wp-content/plugins/wc-post-types/inc/rest-api.php

```diff
@@ -6,51 +6,59 @@
  */
 
 namespace WordCamp\Post_Types\REST_API;
-use WP_Rest_Server;
+use WP_Rest_Server, WP_Post_Type;
 
 defined( 'WPINC' ) || die();
 
 require_once( 'favorite-schedule-shortcode.php' );
 
 /**
- * Add non-sensitive meta fields to the speaker/session REST API endpoints
+ * Add non-sensitive meta fields to WordCamp post type REST API endpoints.
  *
  * If we ever want to register meta for purposes other than exposing it in the API, then this function will
  * probably need to be re-thought and re-factored.
  *
- * @uses wcorg_register_meta_only_on_endpoint()
- *
  * @return void
  */
 function expose_public_post_meta() {
-	$public_session_fields = array(
-		'_wcpt_session_time' => array(
-			'type'   => 'integer',
-			'single' => true,
-		),
-
-		'_wcpt_session_type' => array(
-			'single' => true,
-		),
+	$public_meta_fields = [
+		'wcb_session' => [
+			'_wcpt_session_time' => array(
+				'type'   => 'integer',
+				'single' => true,
+			),
 
-		'_wcpt_session_slides' => array(
-			'single' => true,
-		),
+			'_wcpt_session_type' => array(
+				'single' => true,
+			),
 
-		'_wcpt_session_video' => array(
-			'single' => true,
-		),
-	);
+			'_wcpt_session_slides' => array(
+				'single' => true,
+			),
 
-	wcorg_register_meta_only_on_endpoint( 'post', $public_session_fields, '/wp-json/wp/v2/sessions/' );
+			'_wcpt_session_video' => array(
+				'single' => true,
+			),
+		],
+		'wcb_sponsor' => [
+			'_wcpt_sponsor_website' => array(
+				'single' => true,
+			),
+		],
+	];
 
-	$public_sponsor_fields = array(
-		'_wcpt_sponsor_website' => array(
-			'single' => true,
-		),
-	);
+	foreach ( $public_meta_fields as $post_type => $fields ) {
+		foreach ( $fields as $field_key => $field_args ) {
+			$args = array_merge(
+				$field_args,
+				[
+					'show_in_rest' => true,
+				]
+			);
 
-	wcorg_register_meta_only_on_endpoint( 'post', $public_sponsor_fields, '/wp-json/wp/v2/sponsors/' );
+			register_post_meta( $post_type, $field_key, $args );
```

Comment:
When you view `wp-json/wp/v2/session/N`, does this place the safelisted meta fields at `$session->meta->_wcpt_session_type`, or at `$session->_wcpt_session_type`?

I'm assuming (and hoping) that it's the former. If it's the latter, that would really annoy me, but would probably be a decision that Core made, so probably wouldn't be worth trying to work around anymore.

`MUST` - We might need to consider back-compat concerns, though, since we'd be breaking any clients that rely on that field being under `$session->meta` like it currently is.

Reply:
It's the former, `$session->meta->_wcpt_session_type`. There shouldn't be any backcompat issues.

Reply:
👍